pkg/daemon/daemon.go

```diff
@@ -201,17 +201,28 @@ configRetry:
 	t.BuildInfoLogCxt.WithField("config", configParams).Info(
 		"Successfully loaded configuration.")
 
-	// Make sure the datastore is initialized, otherwise the Syncer may spin, looking for
-	// non-existent resources.
-	for {
-		err := t.DatastoreClient.EnsureInitialized()
-		if err != nil {
-			log.WithError(err).Error("Failed to ensure datastore was initialized")
-			time.Sleep(1 * time.Second)
-			continue
+	// Ensure that, as soon as we are able to connect to the datastore at all, it is
+	// initialized; otherwise the Syncer may spin, looking for non-existent resources.
+	//
+	// But, do this in a background goroutine so as not to block Typha overall; specifically, we
+	// want Typha to report itself as live even if there is an initial problem connecting to the
+	// datastore.
+	//
+	// Note that Typha should cope with intermittent loss of connectivity to the datastore,
+	// because - apart from this EnsureInitialized call here - all of its interaction with the
+	// datastore is via a Syncer, and the Syncer is designed to handle HTTP request errors by
+	// reconnecting.
+	go func() {
+		for {
+			err := t.DatastoreClient.EnsureInitialized()
```

Comment:
If datastore is not available for a period of time, go routine will output quite amount of logs. Is this a concern? Should we limit total amount of logs?

Reply:
I think if that was the case, the main problem would be fixing the availability problem and not the quantity of Felix logs.  Also, please note that the PR here does not change the number of logs generated, I think.

So, I don't think this is a concern for this PR.

Reply:
I see.